src/sdk/pynni/nni/msg_dispatcher.py

```diff
@@ -157,7 +153,21 @@ def handle_trial_end(self, data):
                 self.assessor.trial_end(trial_job_id, data['event'] == 'SUCCEEDED')
         return True
 
+    def _handle_final_metric_data(self, data):
+        '''
+        Call tuner to process final results
+        '''
+        id_ = data['parameter_id']
+        value = data['value']
+        if id_ in _customized_parameter_ids:
+            self.tuner.receive_customized_trial_result(id_, _trial_params[id_], value)
+        else:
+            self.tuner.receive_trial_result(id_, _trial_params[id_], value)
+
     def _handle_intermediate_metric_data(self, data):
+        '''
+        Call assessor to process intermediate results
```

Comment:
same here. You could reference the docstring in xgboost.

Reply:
Thanks, updated.